proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -4,26 +4,54 @@ package determined.experiment.v1;
 option go_package = "github.com/determined-ai/determined/proto/pkg/experimentv1";
 
 import "google/protobuf/struct.proto";
+import "protoc-gen-swagger/options/annotations.proto";
 
-// WorkloadKind defines the kind of workload that should be executed by trial
-// runners.
-enum Unit {
-  // Denotes an unknown Uz.
-  UNIT_UNSPECIFIED = 0;
-  // Unit type "Batches".
-  UNIT_BATCHES = 1;
-  // Unit type "Records".
-  UNIT_RECORDS = 2;
-  // Unit type "Epochs".
-  UNIT_EPOCHS = 3;
+// The representation for an amount of training.
+message TrainingLength {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "units", "length" ] }
+  };
+  // The units for the training length.
+  enum Units {
+    // Zero-value (not allowed).
+    UNITS_UNSPECIFIED = 0;
+    // Indicates a training length is specified in records, samples or another
+    // synonymous unit.
+    UNITS_RECORDS = 1;
+    // Indicates a training length is specified in batches.
+    UNITS_BATCHES = 2;
+    // Indicates a training length is specified in epochs.
+    UNITS_EPOCHS = 3;
+  }
+  // The units the training length is in terms of.
+  Units units = 1;
+  // The value for the training length.
+  int32 length = 2;
+}
+
+// ValidateAfterOperation means the trial should train and validate after
+// training the given length.
+message ValidateAfterOperation {
+  // The length to train before reporting a validation.
+  TrainingLength length = 1;
```

Comment:
You know, after seeing the object-oriented API here, I'm not sure it's any different than having the REST API/RPC take `batches_trained` or whatever when posting a completed validation.  In the version you have here, it's a little clearer that the TrainingLength is being used as an identifier rather for the searcher operation rather than asking for the amount actualy trained.

But I think the python sdk that wraps this will be [OO anyway](https://github.com/rb-determined-ai/determined/blob/sprinkle-spec/sprinkle/low-level/push.py#L72), so I don't think it makes much difference to me how the server-side API asks for it.

Effectively, I am retracting the strong opinions I previously expressed about the REST API (sorry).

Reply:
> You know, after seeing the object-oriented API here, I'm not sure it's any different than having the REST API/RPC take batches_trained or whatever when posting a completed validation

yeh, it's very literally the same thing in a slightly different format. though semantically is a bit different.